Packs/GreatHorn/Integrations/GreatHorn/GreatHorn.py

```diff
@@ -0,0 +1,546 @@
+import json
+import traceback
+from typing import Any, Dict, Optional
+
+import demistomock as demisto  # noqa: F401
+import urllib3
+from CommonServerPython import *  # noqa: F401
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+    """Client class to interact with the service API
+
+    This Client implements API calls, and does not contain any Demisto logic.
+    Should only do requests and return data.
+    It inherits from BaseClient defined in CommonServer Python.
+    Most calls use _http_request() that handles proxy, SSL verification, etc.
+    """
+
+    def get_policy(self, policy_id: Optional[str] = None) -> Dict[str, Any]:
+        if policy_id:
+            return self._http_request(
+                method='GET',
+                url_suffix='/policy/{}'.format(policy_id))
+        else:
+            return self._http_request(
+                method='GET',
+                url_suffix='/policy')
+
+    def set_policy(self, policy_id: str, update_method: str, policy_json: Dict[str, Any]) -> Dict[str, Any]:
+        update_method = update_method.upper()
+        return self._http_request(
+            method=update_method,
+            url_suffix='/policy/{}'.format(policy_id),
+            json_data=policy_json)
+
+    def search_events(self, args: Optional[dict]) -> Dict[str, Any]:
+        return self._http_request(
+            method='POST',
+            url_suffix='/search/events',
+            json_data=args)
```

Comment:
Just to make sure, when do you use the argument `max_fetch`? I might be wrong but it seems like you do not use it to limit the number of incidents that you have got from the API.

Reply:
max_fetch is passed as limit to  gh_get_phish_reports_command and gh_get_quarantine_release_command which use it to set the argument limit which is passed to search_events.

Reply:
What will happen with those incidents that are beyond the limit? For example, if the limit is 100 and there were 102 incidents? Do you save the last 2 incidents somewhere? Do you want to let the user know that there were incidents that are not being displayed?